### PR TITLE
Core: move dimensions out of geomap into app/features

### PR DESCRIPTION
**Pull request by ryantxu**
This moves the "dimension" config and editors out of geomap and into app features so they can more cleanly be shared with:
* geomap
* [scatter plot](https://github.com/grafana/grafana/pull/37423)
* [OHLC](https://github.com/grafana/grafana/pull/36282) 
* [canvas](https://github.com/grafana/grafana/pull/37279) 

This PR takes the dimensions used in https://github.com/grafana/grafana/pull/37279 and makes them useable for core plugins.

We may eventually want the config to be part of `@grafana/data` but the UI elements should stay in core

### Color dimension: 
fixed or based on a field
image

### Scale dimension:
fixed or based on a field percentage (and then scaled by the panel config)
image

image

### Resource dimension:
* mode = "mappings" will take field value and pick an image/icon
* mode = "template" -- use the field value to create a URL
image
image
image

### Text dimension:
image

Questions (but not for this PR)
* currently "value" assumes "last not null" -- maybe we need another config for that?  when it is for a series (ie, canvas vs geomap)

### CODE REVIEW

File: packages/grafana-ui/src/components/OptionsUI/string.tsx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 import React, { useCallback } from 'react';
-import { FieldConfigEditorProps, StringFieldConfigSettings } from '@grafana/data';
+import { StandardEditorProps, StringFieldConfigSettings } from '@grafana/data';
 import { Input } from '../Input/Input';
 import { TextArea } from '../TextArea/TextArea';
 
-export const StringValueEditor: React.FC<FieldConfigEditorProps<string, StringFieldConfigSettings>> = ({
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/OptionsUI/string.tsx:6]: should not be limited to FieldConfig

File: public/app/features/dimensions/editors/ColorDimensionEditor.tsx [hunk 2]
```diff
@@ -5,7 +5,7 @@ import { Select, ColorPicker, useStyles2 } from '@grafana/ui';
 import {
   useFieldDisplayNames,
   useSelectOptions,
-} from '../../../../../../../packages/grafana-ui/src/components/MatchersUI/utils';
+} from '../../../../../packages/grafana-ui/src/components/MatchersUI/utils';
```
Comment (ryantxu) [inline public/app/features/dimensions/editors/ColorDimensionEditor.tsx:8]: this is using unexpored functions from `@grafana/ui`

**Review by lean257 — APPROVED** [review]

**Comment by ryantxu** [comment]
@dprokop -- I'm going to go ahead and merge... I have 4 open PRs that use this :grimacing: keeping them up-to-date is a pain

**Comment by dprokop** [comment]
@ryantxu I've went through the code and tested a bit against https://github.com/grafana/grafana/pull/37423 and this looks great.

**Pull request merged**